artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/mapped/MappedSequentialFile.java

```diff
@@ -110,7 +110,7 @@ public void open(int maxIO, boolean useExecutor) throws IOException {
    @Override
    public boolean fits(int size) {
       checkIsOpen();
-      final long newPosition = this.mappedFile.position() + size;
+      final long newPosition = (long) this.mappedFile.position() + size;
       final boolean hasRemaining = newPosition <= this.mappedFile.length();
```

Comment:
No need for casting. Is just code clutter.

Reply:
It may change semantics, depending on how large the variables are. You are adding together two ints, then assigning to long. Without the cast, you can get overflow on the int range, before you assign to that long. (Which is what the IntLongMath warning is about).

Reply:
```
jshell> Integer.MAX_VALUE + Integer.MAX_VALUE
$2 ==> -2

jshell> (long) Integer.MAX_VALUE + Integer.MAX_VALUE
$3 ==> 4294967294
```

Reply:
Ah yes i forgot about that. Good stuff